Extensions/P2P/B_p2ptools.ts

```diff
@@ -350,52 +381,38 @@ namespace gdjs {
       /**
        * Returns the latest error message.
        */
-      export const getLastError = (): string => {
-        return lastError;
-      };
+      export const getLastError = () => lastError;
 
       /**
        * Returns true once a peer disconnected.
        */
-      export const onDisconnect = (): boolean => {
-        return disconnectedPeers.length > 0;
-      };
+      export const onDisconnect = () => disconnectedPeers.length > 0;
 
       /**
        * Get the ID of the peer that triggered onDisconnect.
        */
-      export const getDisconnectedPeer = (): string => {
-        return disconnectedPeers[disconnectedPeers.length - 1] || '';
-      };
+      export const getDisconnectedPeer = () => disconnectedPeers[0] || '';
 
       /**
        * Returns true once if a remote peer just initiated a connection.
        */
-      export const onConnection = (): boolean => {
-        return connectedPeers.length > 0;
-      };
+      export const onConnection = () => connectedPeers.length > 0;
 
       /**
        * Get the ID of the peer that triggered onConnection.
        */
-      export const getConnectedPeer = (): string => {
-        return connectedPeers[connectedPeers.length - 1] || '';
-      };
+      export const getConnectedPeer = (): string => connectedPeers[0] || '';
 
       gdjs.callbacksRuntimeScenePostEvents.push(() => {
-        for (const i in lastEventData) {
-          if (
-            typeof lastEventData[i] === 'object' &&
-            lastEventData[i].length > 0
-          ) {
-            (lastEventData[i] as string[]).pop();
-          }
+        for (const i in events) {
+          if (!events.hasOwnProperty(i)) continue;
+          events[i].popData();
```

Comment:
nitpicking: `i` is a shorthand for `index`, which means to the reader that this is almost surely an array.
Prefer to name your key as what it represents: the event name, so `for (const eventName in events) { ...`